revpi_gate.c

```diff
@@ -29,6 +29,8 @@ static LIST_HEAD(revpi_gate_connections);
 static DEFINE_MUTEX(revpi_gate_lock);		/* serializes list add/del */
 DEFINE_STATIC_SRCU(revpi_gate_srcu);		/* protects list traversal */
 static DECLARE_WAIT_QUEUE_HEAD(revpi_gate_fini_wq);
+static struct sk_buff_head revpi_gate_rcvq;
+static struct task_struct *revpi_gate_rcv_thread;
```

Comment:
I still favor the solution of using a raw spinlock for process image acess and copying the data to exchange with userspace in a local buffer to avoid callling copy_[from|to]_user() under the lock. That would spare us a new kthread and additional complexity by copying data to/from a queue. However I am also fine with this solution.

Reply:
Please let's stick with the existing solution. I fear we'd have to comb through the entire module and check where we're holding the lock in sleeping context. If we go to these lengths, we may as well rework the whole thing to use lockless access (as alluded to in the commit message).